### PR TITLE
Base for Signer Websockets server

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

File: signer/src/ws_server.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+use std::net::SocketAddr;
+use util::panics::{PanicHandler, OnPanicListener, MayPanic};
+
+/// Signer startup error
+#[derive(Debug)]
+pub enum ServerError {
+	/// Wrapped `std::io::Error`
+	IoError(std::io::Error),
+	/// Other `ws-rs` error
+	Other(ws::Error)
```
Comment (rphmeier) [inline signer/src/ws_server.rs]: maybe call this variant WebSocket?

**Comment by tomusdrw** [comment]
Together with @debris we decieded not to move rpc types to signer create, but rather expose the types from `rpc`.

File: parity/signer.rs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			deps.panic_handler.forward_from(&server);
+			Some(server)
+		},
+	}
+}
+
+#[cfg(not(feature = "ethcore-signer"))]
+pub fn start(conf: Configuration) -> !{
+	if !conf.enabled {
+		return None;
```
Comment (rphmeier) [inline parity/signer.rs]: cannot return, diverging function

File: dapps/src/lib.rs [hunk 3]
```diff
@@ -34,9 +34,9 @@
 //! 	let io = IoHandler::new();
 //! 	io.add_method("say_hello", SayHello);
 //! 	let _server = Server::start_unsecure_http(
-//!		&"127.0.0.1:3030".parse().unwrap(),
-//!		Arc::new(io)
-//!	);
+//!			&"127.0.0.1:3030".parse().unwrap(),
+//!			Arc::new(io)
+//!		);
```
Comment (rphmeier) [inline dapps/src/lib.rs]: indentation

File: parity/main.rs [hunk 4]
```diff
@@ -50,6 +50,11 @@ extern crate ethcore_rpc;
 #[cfg(feature = "dapps")]
 extern crate ethcore_dapps;
 
+#[cfg(feature = "ethcore-signer")]
+extern crate ethcore_signer;
+
+
```
Comment (gavofyork) [inline parity/main.rs]: two superfluous blank lines.

**Pull request merged**